istioctl/pkg/validate/validate.go

```diff
@@ -195,6 +195,17 @@ func (v *validator) validateDeploymentLabel(istioNamespace string, un *unstructu
 	}
 }
 
+// GetTemplateLabels returns spec.template.metadata.labels from Deployment
+func GetTemplateLabels(u *unstructured.Unstructured) map[string]string {
+	if spec, ok := u.Object["spec"].(map[string]interface{}); ok {
+		if template, ok := spec["template"].(map[string]interface{}); ok {
+			m, _, _ := unstructured.NestedStringMap(template, "metadata", "labels")
```

Comment:
Don't ignore the error return from NestedStringMap().  It should be considered an error, not a warning, for validation.

To reproduce a situation where the error is meaningful, I modified _samples/httpbin/httpbin.yaml_, with

```
  template:
    metadata:
      labels:
        app: httpbin
        version: v1
        zero: 0
```

If `err` is ignored, your code prints that there is a problem with missing label `"version"`, which is untrue.

Add a test case for that as well.

Reply:
> Don't ignore the error return from NestedStringMap().

The current semantics from `unstructured` lib is to [return only value](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/unstructured.go#L394-L397) and not an error. I followed the same style which [was used previously to get ](https://github.com/istio/istio/blob/master/istioctl/pkg/validate/validate.go#L191)`Deployment` labels.

> To reproduce a situation where the error is meaningful, I modified samples/httpbin/httpbin.yaml,

Are we going to validate if the labels should accept values other than `string`?

Because `zero: 0` shows `Incorrect type. Expected "string"`.

And is also not a valid `yaml`.
```
$ kubectl create -f httpbin.yaml
Error from server (BadRequest): error when creating "httpbin.yaml":
Deployment in version "v1" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.ObjectMeta: v1.ObjectMeta.Labels:
ReadString: expects " or n, but found 0, error found in #10 byte of ...|","zero":0}},"spec":|..., bigger context ...|{"labels":{"app":"httpbin","version":"v1","zero":0}},"spec":{"containers":[{"image":"docker.io/kenne|...
```

On the other hand, `httpbin` is a sample and not going to be updated by the end-user so this test looks overwork.

Reply:
The labels must be strings.  I think you have a good solution -- I didn't know about `unstructured.NestedStringMap()`.  I am suggeting if NestedStringMap returns an `error` you might want GetTemplateLabels() to also return that error.

Reply:
FYI Earlier today a new framework merged that allows the webhook to issue warnings.

https://github.com/istio/istio/pull/27180

That mechanism won't work for Deployment because it only works for Istio CRs.  You should be aware that soon the TODO on https://github.com/istio/istio/pull/27180/files#diff-baef65426d2d2b26ba35559728c3e783R99 will be implemented and all types will be able to output warning.  We want everything to work well together.

Reply:
> I am suggeting if NestedStringMap returns an error you might want GetTemplateLabels() to also return that error.

Added error handling.